## Actividad Evaluable: Obtención de estadísticas descriptivas
### Pedro Luis Pérez Collado-A01664621


### 1. Carga los datos usando tu lector de csv o con pandas. Es recomendable hacerlo con pandas.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Cargamos los datos
df = pd.read_csv('TC1002S/avocado.csv')   


### 2. Verifica la cantidad de datos que tienes, las variables que contiene cada vector de datos e identifica el tipo de variables.

In [2]:
#Verificando la cantidad de datos
print(df.shape)

(18249, 14)


In [3]:
#Verificar las variables que contiene cada vector
print(df.columns)

Index(['Unnamed: 0', 'Date', 'AveragePrice', 'Total Volume', '4046', '4225',
       '4770', 'Total Bags', 'Small Bags', 'Large Bags', 'XLarge Bags', 'type',
       'year', 'region'],
      dtype='object')


In [4]:
#Identificar el tipo de variable de cada vector
print(df.dtypes)

Unnamed: 0        int64
Date             object
AveragePrice    float64
Total Volume    float64
4046            float64
4225            float64
4770            float64
Total Bags      float64
Small Bags      float64
Large Bags      float64
XLarge Bags     float64
type             object
year              int64
region           object
dtype: object


### 3. Analiza las variables para saber que representa cada una y en que rangos se encuentran. Si la descripción del problema no te lo indica, utiliza el máximo y el mínimo para encontrarlo.

In [5]:
#Analizar las variables para saber que representan y sus rangos
print(df.describe())


         Unnamed: 0  AveragePrice  Total Volume          4046          4225  \
count  18249.000000  18249.000000  1.824900e+04  1.824900e+04  1.824900e+04   
mean      24.232232      1.405978  8.506440e+05  2.930084e+05  2.951546e+05   
std       15.481045      0.402677  3.453545e+06  1.264989e+06  1.204120e+06   
min        0.000000      0.440000  8.456000e+01  0.000000e+00  0.000000e+00   
25%       10.000000      1.100000  1.083858e+04  8.540700e+02  3.008780e+03   
50%       24.000000      1.370000  1.073768e+05  8.645300e+03  2.906102e+04   
75%       38.000000      1.660000  4.329623e+05  1.110202e+05  1.502069e+05   
max       52.000000      3.250000  6.250565e+07  2.274362e+07  2.047057e+07   

               4770    Total Bags    Small Bags    Large Bags    XLarge Bags  \
count  1.824900e+04  1.824900e+04  1.824900e+04  1.824900e+04   18249.000000   
mean   2.283974e+04  2.396392e+05  1.821947e+05  5.433809e+04    3106.426507   
std    1.074641e+05  9.862424e+05  7.461785e+05 

In [6]:
#Incluir en la descripción la mediana
print(df.median())

Unnamed: 0          24.00
AveragePrice         1.37
Total Volume    107376.76
4046              8645.30
4225             29061.02
4770               184.99
Total Bags       39743.83
Small Bags       26362.82
Large Bags        2647.71
XLarge Bags          0.00
year              2016.00
dtype: float64


/var/folders/44/0pfv53sn2hs3ghwtq8p2d6lc0000gn/T/ipykernel_95619/250106321.py:2: FutureWarning: The default value of numeric_only in DataFrame.median is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  print(df.median())


### 4. Basándose en la media, mediana y desviación estádar de cada variable, que conclusiones puedes entregar de los datos.

Que el precio promedio fue de 1.4 dólares, esto debido a que analicé la fuente original y se que esta columna está en dólares, también que el precio mínimo histórico en esta base de datos fue 0.4 dólares y el máximo histórico en esta base de datos es de 3.25 dólares. Que en promedio se vendieron más bolsas pequeñas y que el año 2016 es el que se encuentra en medio de los datos.